Vagrantfile

```diff
@@ -28,20 +28,28 @@ $forwarded_ports = {}
 $subnet = "172.17.8"
 $os = "ubuntu"
 $network_plugin = "flannel"
+
+if File.exist?(CONFIG)
+  require CONFIG
+end
+
+if $kube_master_instances.nil?
+  if $num_instances == 1
+    $kube_master_instances = $num_instances
+  elsif $num_instances > 1 and $num_instances <= 3
+    $kube_master_instances = 2
+  elsif $num_instances > 3
+    $kube_master_instances = 3
+  end
+end
 # The first three nodes are etcd servers
-$etcd_instances = $num_instances
-# The first two nodes are kube masters
-$kube_master_instances = $num_instances == 1 ? $num_instances : ($num_instances - 1)
+$etcd_instances = $kube_master_instances < 3 ? 3 : $kube_master_instances
```

Comment:
i'm not sure about single node setup, but if `num_instances == 1` then `etcd_instances` has incorrect value 3

Reply:
why not just have a `etcd_instances` ?